java_src/foam/dao/LastModifiedAwareDAO.java

```diff
@@ -0,0 +1,23 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.dao;
+
+import foam.dao.*;
+import foam.core.*;
+import foam.nanos.auth.LastModifiedAware;
+import java.time.LocalDateTime;
+
+public class LastModifiedAwareDAO
+  extends ProxyDAO
+  implements LastModifiedAware
+{
+  public FObject put(FObject value) {
+    ((LastModifiedAware)value).lastModified = LocalDateTime.now();
+    super.put(value);
```

Comment:
The delegate could return a different object, so the correct semantics would be:
return super.put(value);